flink-connector/flink/src/main/java/org/apache/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -521,6 +552,11 @@ protected CatalogBaseTable toFlinkTable(Table table) {
           .column(column.name(), column.nullable() ? flinkType.nullable() : flinkType.notNull())
           .withComment(column.comment());
     }
+    Index[] indices = table.index();
```

Comment:
Could you add a check here, the index array contains at more one index, and it should be Primary index.